node-notes-cli/app.js

```diff
@@ -0,0 +1,157 @@
+import { writeFile } from 'node:fs/promises';
+import { readFileSync } from 'node:fs';
+
+// Set up function and variables to handle user inputs
+const action = process.argv[2];
+const arg3 = process.argv[3];
+const arg4 = process.argv[4];
+const legalActions = ['read', 'create', 'update', 'delete'];
+
+function interpretInputs() {
+  switch (action) {
+    case 'read':
+      readNotes();
+      break;
+    case 'create':
+      newNote();
+      break;
+    case 'update':
+      updateNote();
+      break;
+    case 'delete':
+      deleteNote();
+      break;
+    default:
+      console.error(action, 'is not an acceptable action. Please try again with:', legalActions);
+      process.exit(1);
+  }
+}
+
+/**
+ * Reusable functions
+ */
+
+// Catch error handling
+function handleError(err) {
+  console.error(err);
+  process.exit(1);
+}
+
+// Evaluate ID
+async function evaluateId() {
```

Comment:
Very clever error handling. I actually wouldn't have considered doing this. I like it!